pkgs/os-specific/linux/systemd/default.nix

```diff
@@ -695,6 +699,10 @@ stdenv.mkDerivation {
     mv $out/dont-strip-me $out/lib/systemd/boot/efi
   '';
 
+  disallowedReferences = lib.optionals (stdenv.buildPlatform != stdenv.hostPlatform)
+    # 'or p' is for manually specified buildPackages as they dont have __spliced
+    (builtins.map (p: p.__spliced.buildHost or p) finalAttrs.nativeBuildInputs);
```

Comment:
A bit odd we have to interact with `__spliced` directly here, but ugh, fair enough. Thanks for fixing!